packages/base4-rest-api/src/client.js

```diff
@@ -161,6 +171,7 @@ class Base4RestApiClient {
   }
 
   buildUrl({ type, endpoint } = {}) {
+    if (type === 'upload') return `${this.uri}/upload${cleanPath(endpoint)}`;
```

Comment:
Can you make this a “custom” type and simply clean and return whatever endpoint is provided? Otherwise it’s kinda confusing. 🙂